skyportal/handlers/api/spectrum.py

```diff
@@ -408,25 +360,19 @@ def post(self):
         if instrument is None:
             raise ValidationError('Invalid instrument id.')
 
-        groups = []
-        group_ids = json.pop('group_ids', [])
-        for group_id in group_ids:
-            group = Group.query.get(group_id)
-            if group is None:
-                return self.error(f'Invalid group id: {group_id}.')
-            groups.append(group)
-
         # always add the single user group
-        single_user_group = (
-            DBSession()
-            .query(Group)
-            .join(GroupUser)
-            .filter(
-                Group.single_user_group == True,  # noqa
-                GroupUser.user_id == self.associated_user_object.id,
+        single_user_group = self.associated_user_object.single_user_group
+
+        group_ids = json.pop('group_ids', [])
+        if not group_ids:
+            groups = [single_user_group]
+        else:
```

Comment:
should we also allow "all" and handle accordingly here?

Reply:
Thanks, good catch.